packages/plugin-ext/src/main/node/plugins-key-value-storage.ts

```diff
@@ -110,7 +110,16 @@ export class PluginsKeyValueStorage {
 
     private async writeToFile(pathToFile: string, data: KeysToKeysToAnyValue): Promise<void> {
         await fs.ensureDir(path.dirname(pathToFile));
-        await fs.writeJSON(pathToFile, data);
+        const tmpFile = pathToFile + '.tmp';
+        await new Promise((resolve, reject) => {
+            fs.writeJSON(tmpFile, data, {}, error => {
+                if (error) {
+                    reject(error);
+                } else {
+                    fs.rename(tmpFile, pathToFile, resolve);
```

Comment:
callback shouldn't be `resolve` alone but something like

```ts
error => {
  if (error) {
    reject(error);
  } else {
    resolve();
  }
}
```

Alternatively you could use Node's [`promises` fs apis](https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_fs_promises_api) if you find it simplier.